modules/gltf/gltf_document.cpp

```diff
@@ -1976,6 +1972,91 @@ GLTFAccessorIndex GLTFDocument::_encode_accessor_as_vec3(Ref<GLTFState> p_state,
 	return p_state->accessors.size() - 1;
 }
 
+GLTFAccessorIndex GLTFDocument::_encode_sparse_accessor_as_vec3(Ref<GLTFState> p_state, const Vector<Vector3> p_attribs, const Vector<Vector3> p_reference_attribs, const bool p_for_vertex, const GLTFAccessorIndex p_reference_accessor) {
+	if (p_attribs.size() == 0) {
+		return -1;
+	}
+
+	const int element_count = 3;
+	Vector<double> attribs, type_max, type_min;
+	attribs.resize(p_attribs.size() * element_count);
+	type_max.resize(element_count);
+	type_min.resize(element_count);
+
+	Vector<double> changed_indices;
+	Vector<double> changed_values;
+
+	for (int i = 0; i < p_attribs.size(); i++) {
+		Vector3 attrib = p_attribs[i];
+		attribs.write[(i * element_count) + 0] = _filter_number(attrib.x);
+		attribs.write[(i * element_count) + 1] = _filter_number(attrib.y);
+		attribs.write[(i * element_count) + 2] = _filter_number(attrib.z);
+		bool is_different = false;
+		if (i < p_reference_attribs.size()) {
+			is_different = !attrib.is_equal_approx(p_reference_attribs[i]);
+		} else {
+			is_different = !attrib.is_zero_approx();
+		}
+		if (is_different) {
+			changed_indices.push_back(i);
+			changed_values.push_back(_filter_number(attrib.x));
+			changed_values.push_back(_filter_number(attrib.y));
+			changed_values.push_back(_filter_number(attrib.z));
+		}
+		_calc_accessor_min_max(i, element_count, type_max, attribs, type_min);
+	}
+	_round_min_max_components(type_min, type_max);
+
+	if (attribs.size() % element_count != 0) {
+		return -1;
+	}
+
+	Ref<GLTFAccessor> sparse_accessor;
+	sparse_accessor.instantiate();
+	int64_t size = p_state->buffers[0].size();
+	const GLTFType type = GLTFType::TYPE_VEC3;
+	const int component_type = GLTFDocument::COMPONENT_TYPE_FLOAT;
+
+	sparse_accessor->normalized = false;
+	sparse_accessor->count = p_attribs.size();
+	sparse_accessor->type = type;
+	sparse_accessor->component_type = component_type;
+	if (p_reference_accessor < p_state->accessors.size() && p_reference_accessor >= 0 && p_state->accessors[p_reference_accessor].is_valid()) {
+		sparse_accessor->byte_offset = p_state->accessors[p_reference_accessor]->byte_offset;
+		sparse_accessor->buffer_view = p_state->accessors[p_reference_accessor]->buffer_view;
+	}
+	sparse_accessor->max = type_max;
+	sparse_accessor->min = type_min;
+
+	if (changed_indices.size() > 0 && changed_indices.size() < p_attribs.size() / 2) {
+		// If more than half of vertices changed, it is probably not worthwhile to use a sparse accessor.
```

Comment:
this is a bit arbitrary

Reply:
Some check is needed, otherwise we will attempt to include sparse buffer even when everything changed, which will increase file size, not reduce it. So we do need a cutoff to avoid a size regression but I agree the 1/2 was arbitrary.

Do you think I should do a more rigorous calculation? Also, maybe we should support unsigned_short indices here... (spekaing of unsigned_short, is there a reason we don't use it for index buffers on meshes with <64k vertices?)

Reply:
I think that we can do a more rigorous calculation. Let me know if it's a challenge or otherwise exceedingly difficult.